snuba/utils/streams/kafka.py

```diff
@@ -107,10 +125,20 @@ def commit(self) -> Mapping[TopicPartition, int]:
                 retries_remaining -= 1
                 time.sleep(1)
 
-        return {
-            TopicPartition(value.topic, value.partition): value.offset
-            for value in result
-        }
+        offsets: MutableMapping[TopicPartition, int] = {}
+
+        for value in result:
+            # The Confluent Kafka Consumer will include logical offsets in the
+            # sequence of ``TopicPartition`` objects returned by ``commit``.
+            # These are an implementation detail of the Kafka Consumer, so we
+            # don't expose them here.
+            if value.offset in self.LOGICAL_OFFSETS:
```

Comment:
Not sure I get why we can hide these offset from the client. Is there any use case where these offset should be used by the product ?

Reply:
These are returned from `commit` with the value corresponding to the configuration of `auto.offset.reset` for partitions that haven't had any messages consumed yet by this consumer. They don't have any meaning in this context other than "this partition has an offset that wasn't actually committed yet", which is more easily exposed to the caller by just not including the stream in the return value to indicate nothing was committed for that stream because there was nothing to commit.

The commit log itself hasn't included these values since 17791649565326285b505fb1cd5e4224811946a2 which was a fix to an where the logical offset values being published to the commit log. The logical offset values themselves are negative integers — the same datatype as the offset themselves — and caused unnecessary state transitions and warnings in the synchronized consumer when the offset appeared to get rolled back to the negative offset values. One of my goals with this abstraction is to remove those kinds of issues that are caused by the Confluent API design.